### ISSUE TITLE
file/folder deletion and Flash player

### ISSUE THREAD

**Issue by huskelord** [lurkbbs/e621dl#15]
Ok so i tried the posts deletion and everything seems to work great except the deletion of posts and folders themselves (im actually happy for this because I was worried it would get rid of my folders for the db downloads) I ran the program to completion and everything. the posts in the to_blocked_posts gets deleted and the names get added into the blacklist file (blacklisting works completely by the way) but any posts that are in the blacklist file that are already downloaded are not deleted from the other folders they are in. (to help test download gscot and minus8 from my artists all of gscot's animations are also in minus8's animations)

and second issue is off topic i know but i was wondering if you could help me with something 
post 806614 says I can't run it because my flash player is on 11.1 and it needs to be on 11.2 or higher. when I tried to download it it wouldn't work whatsoever. I can run it using internet explorer but i don't really want to do that for every single flash animation since it opens an Internet Explorer window and I would rather just have the program run on its own and not have to open an internet browser especially internet explorer 
Basically was wondering if you knew anyway i could get the most recent flash player downloaded onto my computer so I could run the flash without having to run it on internet explorer

**Comment by lurkbbs** [comment]
Ah, sorry, I should have said that. Pruning are disabled by default. Add this to enable it

```
[Settings]
...
prune_downloads = true
prune_cache = true
```

And for flash, I recommend to download Flash Projector from [here](https://www.adobe.com/support/flashplayer/debug_downloads.html). It's a standalone app, so you have to setup flash files to open with it.

Just in case, short version how to change it, taken from https://techforluddites.com/windows-10-change-the-default-programs-for-opening-files/

![image](https://user-images.githubusercontent.com/38856256/70860465-0d96c800-1f33-11ea-9953-ed1da02aa9c1.png)

![cb1](https://user-images.githubusercontent.com/38856256/70860494-6bc3ab00-1f33-11ea-8f22-15b1f365f3e6.png)

![cb2](https://user-images.githubusercontent.com/38856256/70860515-a3325780-1f33-11ea-92c0-cbab8504b364.png)

![image](https://user-images.githubusercontent.com/38856256/70860520-ae858300-1f33-11ea-8497-cb5c14156bf6.png)

Except look for flash projector in the last window, of course :-)

**Comment by huskelord** [comment]
thanks. don't know if your still here so i don't know if you'll get this before it run it again but do you know if pruninng downloads will get rid of my DB downloads every time since they are not in that config.ini or will it only do so on serches that that are in that paticular config.ini

Edit1: also just got an idea. would you be able to make a favorites folder that works like the blacklist but basically it just adds a folder called favorites in every folder that has images and puts  a copy of any "favorited" posts into the favorit subfolder inside the relevent subfolder if that's to much trouble of makes no sense don't worry about it I can just make a favorite folder on its own it was just an idea that i thougth would be cool to have.

Edit2: Ok major problem becaus I wanted to use the prunning in both my api config and my db config i figured out that they will prun eachother and bassicly delet anything that isn't in the serch cryteria for whatever config is running at the moment i can probably get away with only pruning during the api but that will just mean all my db downloads get pruned every time it runs and need to be redownloaded/copied and thats just a pain in the but since it takes a while for the program to run through every single post plus pruning does'nt get rid of empty folders just the images which is pretty anoying in of iteself
so bassicly i need a way to eather combine both of my configs into one with me being able to api download the inital part and db download the second so it doesn't prun anything that doesn't need it or it needs to have a way to keep the two downloads from pruning anyting else but i would like have the db be able to prun itself without me having to worry about it pruning it's source material since db will only grab from alrady downloaded posts

I'll try running the progam with the db wihtout prun at first and see it it works once it has downloaded everything it can and do some more testing to figrue it all out.

**Comment by lurkbbs** [comment]
About favorites, I don't understand. Do you want to make favorites folder that should be available from all other subfolders, like a shortcut to favorites in all of the folders?

> will just mean all my db downloads get pruned every time it runs and need to be

OK, that is a problem. Fastest workaround is pruning downloads on first run (api) and pruning cache on second run (db). This means a lot of files will be removed from downloads only to be copied from the cache on second run. Then, when only relevant files are in downloads, we can prune cache from all irrelevant.

The problem is removing a lot of files will take a while, but not as much as redownloading.
Second shortcut, is to make second config to fully include first, e.g first is

```
[Species]
subfolders = feline

```

and second

```
[Species]
subfolders = feline canine

```

That way you can first download new files, than remaining with some slowdown to check files that are already downloaded. A bit more hassle, but a bit faster.

Third option is to wait a bit, I'll try to make some workaround for multiple scripts, than make multiple config runs either totally unneeded or a part of a workflow, without the script, during next two weeks at most.

As for empty folders, I'll fix it sometime during next week, before the weekend.

**Comment by lurkbbs** [comment]
OK, quick workaround, replace both executable and the script:
[2.zip](https://github.com/lurkbbs/e621dl/files/3965177/2.zip)

And disable pruning for the first config.

**Comment by huskelord** [comment]
this one is done i think so if you wan't to close it unless im forgetting something

**Comment by lurkbbs** [comment]
It's not done until actual release. Then I'll definitely close it

**Comment by lurkbbs** [comment]
And closed!
